frontend/src/utils/hooks/operations.ts

```diff
@@ -44,13 +45,15 @@ const PREVIEWBASEURL = api.routes.PREVIEW_SINGLE_DATASET;
 
 const handleDataResult = (status: number, data: OperationDataResult): FetchResponse => {
   if (status === 200 || (status === 201 && data)) {
+    if (data.results && data.results.length && data.results[0].error) {
+      return { status, error: data.results[0].error as string };
+    }
+
     return { data: data.results, status };
   } else if (status === 401) {
     setToken('');
 
     return { status, error: 'invalid token' };
-  } else if (data.results && data.results.length && data.results[0].error) {
-    return { status, error: data.results[0].error as string };
```

Comment:
I understand copying it, but are you sure this can't happen? I'm sure this was meant to handle `any other result` that's not status 200, 201 or 401 ... so I think you'd best be safe and leave this.